File: csharp-code/Inheritance/MultilevelInheritance.cs
using System;
 
namespace Hello
{
    public class Carnivore //base class or parent class
    {
        protected String name;
        
        public String getType()
        {
            return $"{this.name} is a Carnivore ";
        }
    }
    
    public class Animal:Carnivore // derived class or child class
    {
        public Animal(String name)
        {
            base.name=name; // base is similar to super in java
        }
        
        public String getBaseInfo()
        {
            return getType();
        }
    }
    
    public class Tiger:Animal
    {
        private string character;
        public Tiger(String name,String character):base(name)
        {
            this.character=character;
        }
        public String getDetails()
        {
            return getBaseInfo()+" "+this.character;
        }
        
    }
	public class Hello
	{
	    /*
	        C# Multilevel Inheritance
	        
	        output
	        
	        Tiger is a Carnivore  Solitary apex predator
	        
	    */
	    
	    
		public static void Main (string[] args)
		{
            Tiger tiger=new Tiger("Tiger","Solitary apex predator");
            Console.WriteLine(tiger.getDetails());
		}
	}
}

File: Java-notes/Inheritance/MultilevelInheritance.java
/*
    Multi Level Inheritance 

    *** output ***

    Apex Predator and fearless
    Panthera tigris is a carnivore

*/

class Carnivore //base class or parent class
{
    String name ;

    public String getType()
    {
        return this.name+" is a carnivore";
    }
}

class Animal extends Carnivore // derived class or child class
{
    String character;
    public Animal(String name,String character)
    {
        super.name = name;
        this.character  = character;
    }

    public String getDetails()
    {
        return this.character;
    }
}

class Tiger extends Animal
{
    public Tiger(String name,String character)
    {
        super(name,character); // calling parent class Constructore
    }
}

public class MultilevelInheritance
{
    public static void main(String[] args)
    {
        Tiger tiger = new Tiger("Panthera tigris","Apex Predator and fearless");

        System.out.println(tiger.getDetails());

        System.out.println(tiger.getType());

    }
}